src/main/java/net/tropicraft/core/common/entity/passive/EntityKoaBase.java

```diff
@@ -672,6 +707,78 @@ public void setFightingItem() {
         this.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(ItemRegistry.dagger));
     }
 
+    public void monitorHomeVillage() {
+        if (villageDimID != INVALID_DIM) {
+            //if (villageID != -1) {
+
+                //if not in home dimension, full reset
+                if (this.world.provider.getDimension() != villageDimID) {
+                    dbg("koa detected different dimension, zapping memory");
+                    zapMemory();
+                    addPotionEffect(new PotionEffect(MobEffects.SLOWNESS, 5));
+                }
+            //}
+        }
+
+        findOrCreateNewVillage();
+    }
+
+    public void findOrCreateNewVillage() {
+        if ((world.getTotalWorldTime()+this.getEntityId()) % (20*5) != 0) return;
+
+        if (getVillage() == null) {
+            TownKoaVillage village = getVillageWithinRange(128);
+            if (village != null) {
+                dbg("Koa found a village to join: " + village.locationID);
+                this.setVillageAndDimID(village.locationID, village.dimID);
+            } else {
+                //check we have a chest locked in
+                TileEntity tile = world.getTileEntity(getHomePosition());
+                if ((tile instanceof TileEntityChest)) {
```

Comment:
probably need a tile != null check here

Reply:
should be fine without, wont NPE for instance check, and i only use if its TileEntityChest